dev/src/main/java/org/eclipse/codewind/intellij/ui/module/CustomOptionsStep.java

```diff
@@ -0,0 +1,145 @@
+                    message("CodewindErrorMsg");
+                String labelText = message("CodewindStatusError", msg);
+                label.setText(labelText);
+                button.setText(message("RefreshCodewindStatus"));
+                removeActionListeners(button);
+                button.addActionListener(e -> {
+                    CodewindManager.getManager().refreshInstallStatus();
+                    configurePage();
+                });
+            } else if (status.isInstalled()) {
+                label.setText(message("CodewindNotStarted"));
+                button.setText(message("StartCodewind"));
+                removeActionListeners(button);
+                button.addActionListener(e -> {
+                    CodewindManager.getManager().refreshInstallStatus();
+                    if (CodewindManager.getManager().getInstallStatus().isStarted()) {
+                        CoreUtil.openDialog(CoreUtil.DialogType.INFO, message("CodewindLabel"), message("CodewindStarted"));
+                    } else {
+                        ProgressManager.getInstance().run(new StartCodewindTask(this::onStart));
+                    }
+                });
+            } else if (status.hasInstalledVersions()) {
+                label.setText(message("CodewindNotInstalled"));
+                button.setText(message("UpgradeCodewind"));
+                removeActionListeners(button);
+                button.addActionListener(e -> {
+                    CodewindManager.getManager().refreshInstallStatus();
+                    if (CodewindManager.getManager().getInstallStatus().isStarted()) {
+                        CoreUtil.openDialog(CoreUtil.DialogType.INFO, message("CodewindLabel"), message("CodewindStarted"));
+                    } else {
+                        ProgressManager.getInstance().run(new UpgradeCodewindTask(this::onStart));
+                    }
+                });
+            } else {
+                label.setText(message("CodewindNotInstalled"));
+                button.setText(message("InstallCodewind"));
+                removeActionListeners(button);
+                button.addActionListener(e -> {
+                    CodewindManager.getManager().refreshInstallStatus();
+                    if (CodewindManager.getManager().getInstallStatus().isStarted()) {
+                        CoreUtil.openDialog(CoreUtil.DialogType.INFO, message("CodewindLabel"), message("CodewindStarted"));
+                    } else {
+                        ProgressManager.getInstance().run(new InstallCodewindTask(this::onStart));
+                    }
+                });
+            }
+        } catch (Exception e) {
+            Throwable cause = Logger.unwrap(e);
+            Logger.logWarning(cause);
+        }
+    }
+
+    @Override
+    public JComponent getComponent() {
+        return panel;
+    }
+
+    @Override
+    public void updateDataModel() {
+        // Need to override because it's abstract in superclass
```

Comment:
We could put the following in updateDataModel() to fulfill the purpose of the method:

CodewindManager.getManager().refreshInstallStatus();

Reply:
Changes made